### PR TITLE
docs: pipeline rework ADR

### PR DESCRIPTION
**Pull request by phbelitz**
For now, only an ADR for a potetial pipeline rework.

## Checklist

- [ ] PR is rebased to/aimed at branch `develop`
- [ ] PR follows [Contributing Guide](https://github.com/sse-secure-systems/connaisseur/blob/master/docs/CONTRIBUTING.md)
- [ ] Added tests (if necessary)
- [ ] Extended README/Documentation (if necessary)
- [ ] Adjusted versions of image and Helm chart in `Chart.yaml` (if necessary)

### CODE REVIEW

**Review by Starkteetje — CHANGES_REQUESTED** [review]

File: docs/adr/ADR-10_pipeline.md [hunk 1]
```diff
@@ -0,0 +1,128 @@
...
+            - ~~install cosign~~  *// not worth the effort*
+            - setup docker
+            - login registry
+            - generate tags
+            - build + push
+            - ~~sbom~~  *// not worth the effort*
+            - ~~sign~~  *// not worth the effort*
+            - ~~verify~~  *// not worth the effort*
+            - ~~upload pub key~~  *// not worth the effort*
+            - ~~print summary~~  *// no one is looking at this*
```
[the remark this answers is not in the document]
Comment (phbelitz) [reply docs/adr/ADR-10_pipeline.md:96]: If you have seen value in this at some point we can leave it. I didn't, so I would have kicked it out, but I also have no hard feeling towards keeping it.

File: docs/adr/ADR-10_pipeline.md [hunk 2]
```diff
@@ -0,0 +1,128 @@
...
+    - compliance
+        - (ossf [scoring for open source projects])  *// only on push, not pr*
```
[the remark this answers is not in the document]
Comment (phbelitz) [reply docs/adr/ADR-10_pipeline.md:98]: well then let's drop it entirely.

File: docs/adr/ADR-10_pipeline.md [hunk 3]
```diff
@@ -0,0 +1,128 @@
...
+- redundant jobs: scanning of the helm chart is done at least thrice. why? 
+- unnecessary jobs: signing images inside the pipeline is nice and all, but adds a lot of complexity for very little gains. same is true for creating sboms, especially during development pipelines. 
+
+## Considered Options
+
+### Option 1
+
+Some general things first.
+
+1. Let the workflows be in the `workflow` directory and put the reusable parts into it's own (e.g. `reusable` or `components` :shrug:).
```
[the remark this answers is not in the document]
Comment (phbelitz) [reply docs/adr/ADR-10_pipeline.md:76]: That sucks. I crossed that point.

File: docs/adr/ADR-10_pipeline.md [hunk 4]
```diff
@@ -0,0 +1,128 @@
...
+            - ~~upload pub key~~  *// not worth the effort*
+            - ~~print summary~~  *// no one is looking at this*
+    - compliance
+        - (ossf [scoring for open source projects])  *// only on push, not pr*
+        - dependency review [dependency scanning]
+        - check-commit-message [linting for commit messages]
+    - unit-test [unit tests :shrug:]
+    - sast
+        - ~~checkov [helm chart scanning]~~  *// we have enough helm scanning*
+        - (codeql [GitHub's "industry-leading" semantic code analysis engine])  *// only on push, not pr as this is by far the slowest*
```
[the remark this answers is not in the document]
Comment (phbelitz) [reply docs/adr/ADR-10_pipeline.md:104]: Again. If you have a strong feeling this provides value during development, we can keep it. I would have just thrown out everything that didn't seem super necessary to me (in this case it's not even throwing out, but do it less frequently, without loosing its value entirely).

**Comment by Starkteetje** [comment]
We'll likely work on the pipeline and streamline it, but not in the grand scale that we'll use an ADR for

**Pull request closed without merging**